nixos/modules/profiles/hardened.nix

```diff
@@ -19,9 +19,6 @@ with lib;
 
   nix.settings.allowed-users = mkDefault [ "@users" ];
 
-  environment.memoryAllocator.provider = mkDefault "scudo";
```

Comment:
It's not because it's crashing your specific program you should remove it. I would advise overriding it locally and keep it until someone takes maintenance of the hardened profile.